main/src/ca/uwaterloo/flix/util/ParOps.scala

```diff
@@ -17,30 +17,37 @@
 package ca.uwaterloo.flix.util
 
 import java.util
-import java.util.concurrent.{Callable, Executors}
-
+import java.util.concurrent.{Callable, CountDownLatch, Future}
 import scala.jdk.CollectionConverters._
-import scala.collection.parallel._
 import ca.uwaterloo.flix.api.Flix
 
+import scala.collection.mutable.ArrayBuffer
+
 object ParOps {
 
   /**
     * Apply the given function `f` to each element in the list `xs` in parallel.
     */
   @inline
   def parMap[A, B](xs: Iterable[A])(f: A => B)(implicit flix: Flix): Iterable[B] = {
-    // Build the parallel array.
-    val parArray = xs.toParArray
-
-    // Configure the task support.
-    parArray.tasksupport = flix.forkJoinTaskSupport
+    val size = xs.size
+    val out = ArrayBuffer.fill(size)(null.asInstanceOf[B])
```

Comment:
Could this just be Array.new(size)?

Reply:
👍

Reply:
Same problem as [this](https://github.com/flix/flix/pull/6615#discussion_r1382553948) unfortunately

Reply:
Using a class tag is fine, if that is the only problem.

Reply:
Alternatively use an array of objects and then cast